torch/_export/serde/serialize.py

```diff
@@ -1488,6 +1524,110 @@ def deserialize_operator(self, serialized_target: str):
                 target = getattr(target, name)
         return target
 
+    def deserialize_sym_expr(self, s: SymExprNode, hint: Optional[int] = None, force_int: Optional[bool] = True) -> Union[sympy.Expr]:
+        if (base := s.base) is not None:
+            if base.type == "as_bool":
+                assert isinstance(base.value, bool)
+                return base.value
+            elif base.type == "as_float":
+                assert isinstance(base.value, float)
+                return base.value
+            elif base.type == "as_int":
+                assert isinstance(base.value, int)
+                return base.value
+            else:
+                assert base.type == "as_symbol"
+                assert isinstance(base.value, str)
+                if base.value in self.symbol_name_to_symbol:
+                    s = self.symbol_name_to_symbol[base.value]
+                else:
+                    s = sympy.Symbol(base.value, integer=force_int)
+                    self.symbol_name_to_symbol[base.value] = s
+                assert isinstance(s, sympy.Symbol)
+                if (
+                    hint is not None
+                    and not s in self.shape_env.var_to_val
+                ):
+                    self.shape_env.add_var_to_val(s, hint)
+                if vr := self.symbol_name_to_range.get(base.value):
+                    self.shape_env.constrain_symbol_range(
+                        s,
+                        compiler_min=vr.lower,  # type: ignore[arg-type]
+                        compiler_max=vr.upper,  # type: ignore[arg-type]
+                    )
+                return s
+        else:
+            assert s.target is not None
+            *_module, _qualname = s.target.split(".")
+            node_cls = getattr(sys.modules[".".join(_module)], _qualname)
```

Comment:
@zhxchen17 @avikchaudhuri any thoughts on this? storing the node target as the python class name (similar to nn_module_stack).

An alternative is to keep a mapping of known functions <-> names, something like
```
{
    "mul": sympy.Mul,
    "add": sympy.Add,
    ...
}
```
But that requires constant updating if we run into new functions. It potentially is safer though, and easier to catch if some day sympy changes their function names/locations in library (we'd break running this file & upgrading sympy pin, instead of when we try to deserialize something).

Reply:
Hmm, I doubt the locations will change much otherwise any user of sympy will also break? Do the class names here resolve to public paths or private paths (since sympy has many aliases and deep directory structure)?